### PR TITLE
TGUI Communications Console fixes

### LINKED ISSUE #6110: Formatting doesn't work on the new communications console TGUI.
image

### PR DESCRIPTION
**Pull request by KubeRoot**
## About The Pull Request

Fixes https://github.com/BeeStation/BeeStation-Hornet/issues/6110

Ports the following PRs:
- https://github.com/tgstation/tgstation/pull/54765
- https://github.com/tgstation/tgstation/pull/62056
  - Heavily modified to match https://github.com/BeeStation/BeeStation-Hornet/pull/2920
- **Only small parts of** https://github.com/tgstation/tgstation/pull/58389
  - Ported the `can_buy_this_shuttle` proc and usages of it to facilitate easier porting of illegal shuttles.

Fixes the following issues:
- HTML formatting didn't work in messages displayed in the comms console. Now the formatting uses custom sanitisation, restricted to safe tags only.
- I previously accidentally removed the illegal shuttle restrictions - I didn't remember we had it, and didn't notice it since the port was a full system replacement. I have now implemented it back in, based on tg's version, modified to fit our existing code.
- I fucked up. No, really. I left in debug code that restricted the shuttles available in the console *to the 6 cheapest shuttles only.* I believe I put the debug code in when trying to figure out why scrolling was broken.
- The tooltip on the shuttle purchase tab wasn't working - this is because I converted it from buttons to tabs, which do not have built-in tooltips, and forgot to code in a tooltip for it.
- There is a check in the code that prevents you from opening the shuttle purchase menu if you cannot buy shuttles. However, the check is inverted, which means it instead prevents you from *leaving* the shuttle purchase menu. This issue comes from upstream, but presumably wasn't noted due to the button being completely disabled when shuttles cannot be purchased.

## Why It's Good For The Game

Bugfixes are good, accidentally stealthremoving a feature is very bad.

## Testing Photographs and Procedure

<details>

<summary>Screenshots&Videos</summary>

image

image

image

</details>

## Changelog
:cl: KubeRoot, prodirus, Mothblocks, Frostahr
add: Reintroduced the accidentally removed illegal shuttle restriction. Illegal shuttles cannot be normally purchased, consoles need to be hacked or emagged to purchase them. Emagging a console will additionally allow you to log into console without access, while obscuring your real name.
fix: The communications shuttle now supports HTML formatting in messages.
fix: Fixes an embarrasing mistake where the communications console was hardcoded to only list the first 6 shuttles for purchase.
fix: Fixes minor issues with the communications console interface.
/:cl:

### CODE REVIEW

File: code/game/machinery/computer/communications.dm [hunk 1]
```diff
@@ -194,6 +194,8 @@
 			var/datum/map_template/shuttle/shuttle = locate(params["shuttle"]) in shuttles
 			if (!istype(shuttle))
 				return
+			if (!can_purchase_this_shuttle(shuttle))
```
Comment (Sinestia) [inline code/game/machinery/computer/communications.dm:197]: this functions name stands out, may be best as just can_purchase()

**Review by Sinestia — APPROVED** [review]
reverterino reverterino'd

looks good

File: code/game/machinery/computer/communications.dm (hunk 1 above)
Comment (KubeRoot) [reply code/game/machinery/computer/communications.dm:197]: I took the proc from tg, in no small part to help with future compatibility - keeping the name the same will mean less trouble down the line porting other communications console things from tg.

Comment (Sinestia) [reply code/game/machinery/computer/communications.dm:197]: Very well then 👌

**Review by Archanial — APPROVED** [review]

**Pull request merged**